src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -459,6 +491,34 @@ internal static unsafe int CompareOrdinalIgnoreCase(string strA, int indexA, int
                     length--;
                 }
 
+                if (CultureData.InvariantMode)
+                {
+                    while (length != 0)
```

Comment:
Is the sole difference between this section and the one above it that this one doesn't limit itself to just ASCII chars, so that we don't call to the OS in invariant mode?  Consider a) adding a comment, and b) moving this out into a separate function.